src/aap_eda/api/views/activation.py

```diff
@@ -320,6 +323,10 @@ def disable(self, request, pk):
     @action(methods=["post"], detail=True, rbac_action=Action.RESTART)
     def restart(self, request, pk):
         activation = get_object_or_404(models.Activation, pk=pk)
+
+        if str(activation.status) == ActivationStatus.DELETING.value:
```

Comment:
@hsong-rh Would it make sense to make a small function called _check_deleting and raise a 409 exception if its in deleting status.

```
def check_deleting(activation):
    if str(activation.status) == ActivationStatus.DELETING.value:
       raise ApiException(detail="Object is being deleted",code=409)
```
 https://github.com/encode/django-rest-framework/blob/5c07060ce0dcbdad1e38be8a509f5c23b6e9fc63/rest_framework/exceptions.py#L108